feedstock/recipe.py

```diff
@@ -1,43 +1,69 @@
-import pandas as pd
-from datetime import datetime, date
+from datetime import datetime, date, timedelta
 from pangeo_forge_recipes.patterns import ConcatDim, MergeDim, FilePattern
 from pangeo_forge_recipes.recipes import XarrayZarrRecipe
 
-# RIOPS is run every 6 hours (at 00, 06, 12, and 18 UTC).  
-# but, for the moment, let's use today's date and assume that when this recipe is being run,
-# the 00UTC run for today is already available!
-start_date = date.today()
-time00 = datetime.min.time()
-start_time = datetime.combine(start_date, time00)
+# RIOPS is run every 6 hours (at 00, 06, 12, and 18 UTC).
+# but, MSC usually only has 2-days-worth of data on the datamart.
+# Thus, we use yesterday's date to ensure the feedstock can run
+START_DATE = datetime.combine(date.today() - timedelta(days=1), datetime.min.time())
+YYYYMMDD = START_DATE.strftime("%Y%m%d")
+RIOPS_PS_BASE_URL = (
+    "https://dd.weather.gc.ca/model_riops/netcdf/forecast/polar_stereographic"
+)
+
 
 def make_url(variable, time):
+    """Currently hardcoded for 2D variables run at 00Z"""
+    fh = f"{time:03.0f}"
+    # vertical level (SFC or DBS-0.5m for 2D, DBS-all for 3D)
+    if variable in ["VOTEMPER", "VOSALINE", "VOZOCRTX", "VOMECRTY"]:
+        level = "DBS-0.5m"  # Depth-Below-Surface 0.5m
+    else:
+        level = "SFC"  # Surface
+    url = f"{RIOPS_PS_BASE_URL}/2d/00/{fh}/{YYYYMMDD}T00Z_MSC_RIOPS_{variable}_{level}_PS5km_P{fh}.nc"
+    return url
 
```

Comment:
@m9brady, apache beam is complaining about the `RIOPS_PS_BASE_URL`. One option is to move this variable inside the `make_url` function body:



```python
  File "/usr/local/lib/python3.9/dist-packages/pangeo_forge_recipes/executors/beam.py", line 40, in exec_stage
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pangeo_forge_recipes/recipes/xarray_zarr.py", line 155, in cache_input
    fname = config.file_pattern[input_key]
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pangeo_forge_recipes/patterns.py", line 219, in __getitem__
    fname = self.format_function(**format_function_kwargs)
  File "/tmp/tmpmn6sfni2/feedstock/recipe.py", line 23, in make_url
NameError: name 'RIOPS_PS_BASE_URL' is not defined [while running 'Start|cache_input|Reshuffle_000|prepare_target|Reshuffle_001|store_chunk|Reshuffle_002|finalize_target|Reshuffle_003/cache_input/Execute-ptransform-56']
```

Reply:
Will do - should I do the same for the other globals too? `[START_DATE, YYYYMMDD]`?

Reply:
Was still able to build and read locally as of https://github.com/pangeo-forge/riops-feedstock/pull/4/commits/20431b2048a861ebb1dbeec12df9181b04bd39df